File: app/src/main/java/com/admin/theapp/utils/mappers/HotelToHotelModelMapper.java
package com.admin.theapp.utils.mappers;

import android.support.annotation.NonNull;

import com.admin.theapp.Hotel;
import com.admin.theapp.model.HotelModel;
import com.theapp.tools.Mapper;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class HotelToHotelModelMapper extends Mapper<Hotel, HotelModel> {

    @Inject
    HotelToHotelModelMapper() { // for dagger2
    }

    @NonNull
    @Override
    public HotelModel map(@NonNull Hotel hotel) {
        return new HotelModel(
                hotel.getId(),
                hotel.getName(),
                hotel.getAddress(),
                hotel.getStars(),
                hotel.getDistance(),
                hotel.getImageName(),
                hotel.getSuitesAvailability(),
                hotel.getLat(),
                hotel.getLon()
        );
    }
}

File: app/src/main/java/com/admin/theapp/utils/mappers/HotelToHotelModelMapper.kt
package com.admin.theapp.utils.mappers

import com.admin.theapp.Hotel
import com.theapp.entities.HotelModel
import com.theapp.tools.Mapper

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class HotelToHotelModelMapper @Inject constructor() : Mapper<Hotel, HotelModel>() {

    override fun map(from: Hotel) =
            HotelModel(from.id, from.name, from.address, from.stars, from.distance,
                    from.imageName, from.suitesAvailability, from.lat, from.lon)
}
